include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -392,46 +407,82 @@ inline ONNXTensorElementDataType TensorTypeAndShapeInfo::GetElementType() const
 }
 
 inline size_t TensorTypeAndShapeInfo::GetDimensionsCount() const {
-  return OrtGetNumOfDimensions(p_);
+  return OrtGetDimensionsCount(p_);
 }
 
 inline void TensorTypeAndShapeInfo::GetDimensions(int64_t* values, size_t values_count) const {
   OrtGetDimensions(p_, values, values_count);
 }
 
 inline std::vector<int64_t> TensorTypeAndShapeInfo::GetShape() const {
-  std::vector<int64_t> output;
-  output.resize(GetDimensionsCount());
-  GetDimensions(output.data(), output.size());
-  return output;
+  std::vector<int64_t> out;
+  out.resize(GetDimensionsCount());
```

Comment:
Why not just `std::vector<int64_t> out(GetDimensionsCount(), 0);`

Reply:
Good idea!